### PR TITLE
Added types to libertem.api.Context

### PR DESCRIPTION
**Pull request by Iamshankhadeep**
Related to #506 

## Contributor Checklist:

* [x] I have added myself to [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [ ] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [ ] I have added/updated documentation for all user-facing changes
* [ ] I have added/updated test cases

### CODE REVIEW

File: src/libertem/api.py [hunk 1]
```diff
@@ -554,7 +557,7 @@ def create_pick_analysis(self, dataset, x: int, y: int = None, z: int = None):
         parameters = {name: loc[name] for name in ['x', 'y', 'z'] if loc[name] is not None}
         return PickFrameAnalysis(dataset=dataset, parameters=parameters)
 
-    def run(self, job: Union[Job, Analysis], roi=None):
+    def run(self, job: Union[Job, Analysis], roi=None) -> np.ndarray:
```
Comment (sk1p) [inline src/libertem/api.py]: The return type of this method is not always `ndarray`, as noted in the docstring. If you want to express something like "either X or Y", you can use the `Union` type, see also [the official documentation](https://docs.python.org/3/library/typing.html#typing.Union). The whole page on typing should be of interest for fixing this issue

File: src/libertem/api.py [hunk 2]
```diff
@@ -602,7 +605,7 @@ def run(self, job: Union[Job, Analysis], roi=None):
             return analysis.get_results(out)
         return out
 
-    def run_udf(self, dataset: DataSet, udf: UDF, roi=None, progress=False):
+    def run_udf(self, dataset: DataSet, udf: UDF, roi=None, progress=False) -> BufferWrapper:
```
Comment (sk1p) [inline src/libertem/api.py]: Closely re-read the docstring again: `BufferWrapper` is not the correct type here.

File: src/libertem/api.py [hunk 3]
```diff
@@ -635,7 +638,7 @@ def run_udf(self, dataset: DataSet, udf: UDF, roi=None, progress=False):
         """
         return UDFRunner(udf).run_for_dataset(dataset, self.executor, roi, progress=progress)
 
-    def map(self, dataset, f, roi=None):
+    def map(self, dataset, f, roi=None) -> BufferWrapper:
```
Comment (sk1p) [inline src/libertem/api.py]: Same comment as for `run_udf` applies here

File: src/libertem/api.py [hunk 4]
```diff
@@ -95,7 +96,7 @@ def load(self, filetype: str, *args, **kwargs) -> DataSet:
     load.__doc__ = load.__doc__ % {"types": ", ".join(filetypes.keys())}
 
     def create_mask_job(self, factories, dataset, use_sparse=None,
-                        mask_count=None, mask_dtype=None, dtype=None):
+                        mask_count=None, mask_dtype=None, dtype=None) -> np.ndarray:
```
Comment (sk1p) [inline src/libertem/api.py]: `np.adarray` is not the correct type here, please compare with the docstring. You can also have a look at the code of the method itself: find the `return` statements and have a look at what kind of object is returned

File: src/libertem/api.py (hunk 3 above)
Comment (Iamshankhadeep) [reply src/libertem/api.py]: After running Context.map() in jupyter I am getting return type as  libertem.common.buffers.BufferWrapper.

Comment (sk1p) [reply src/libertem/api.py]: Ah, I was wrong here, `BufferWrapper` is correct!

File: packaging/creators.json [hunk 5]
```diff
@@ -1,113 +1,120 @@
...
+    "contribution": "Fixed issue 267"
+  },
+  {
+    "displayname": "Shankhadeep Dey",
+    "authorname": "Dey, Shankhadeep",
+    "affiliation": "Siliguri Institute of Technology",
+    "github": "iamshankhadeep",
+    "contribution": "Added types to libertem.api.Context #643"
+  }
+]
```
Comment (sk1p) [inline packaging/creators.json]: Thanks for adding yourself to `creators.json`! It is a bit problematic though to re-indent the whole file. This makes it harder for us to review the changes, and may cause merge conflicts for other contributors, or maybe even yourself if you need to rebase this PR to the current master at a later point.

File: src/libertem/api.py [hunk 6]
```diff
@@ -602,7 +607,7 @@ def run(self, job: Union[Job, Analysis], roi=None):
             return analysis.get_results(out)
         return out
 
-    def run_udf(self, dataset: DataSet, udf: UDF, roi=None, progress=False):
+    def run_udf(self, dataset: DataSet, udf: UDF, roi=None, progress=False) -> dict:
```
Comment (sk1p) [inline src/libertem/api.py]: I would prefer a more concrete type here - what are the key and value types of this `dict`?

File: src/libertem/api.py [hunk 7]
```diff
@@ -554,7 +559,7 @@ def create_pick_analysis(self, dataset, x: int, y: int = None, z: int = None):
         parameters = {name: loc[name] for name in ['x', 'y', 'z'] if loc[name] is not None}
         return PickFrameAnalysis(dataset=dataset, parameters=parameters)
 
-    def run(self, job: Union[Job, Analysis], roi=None):
+    def run(self, job: Union[Job, Analysis], roi=None) -> Union[np.ndarray, AnalysisResultSet]:
```
Comment (sk1p) [inline src/libertem/api.py]: Looks good!

File: packaging/creators.json (hunk 5 above)
Comment (Iamshankhadeep) [reply packaging/creators.json]: Sorry, it was my mistake I did not notice it.

**Pull request merged**